### PR TITLE
feat: add ml.metrics.pairwise.cosine_similarity function

### PR DESCRIPTION
**Pull request by GarrettWu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/sklearn/metrics/pairwise.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        K(X, Y) = <X, Y> / (||X||*||Y||)
+
+    Args:
+        X (Series or single column DataFrame of array of numeric type):
+            Input data.
+        Y (Series or single column DataFrame of array of numeric type):
+            Input data. X and Y are mapped by indexes, must have the same index.
+
+    Returns:
+        DataFrame with columns of X, Y and cosine_similarity
```
Comment (ashleyxuu) [inline third_party/bigframes_vendored/sklearn/metrics/pairwise.py:26]: Add the type hint as well: `bigframes.dataframe.DataFrame: DataFrame with columns of X, Y and cosine_similarity`

File: bigframes/ml/core.py [hunk 2]
```diff
@@ -84,6 +62,59 @@ def _apply_sql(
 
         return df
 
+    def distance(
```
Comment (ashleyxuu) [inline bigframes/ml/core.py:65]: Can we add docstring here too?

File: tests/system/small/ml/test_metrics_pairwise.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+
+from bigframes.ml import metrics
+import bigframes.pandas as bpd
+
+
+def test_cosine_similarity():
```
Comment (ashleyxuu) [inline tests/system/small/ml/test_metrics_pairwise.py:22]: I assume this ML.DISTANCE COSINE in BQML is different from sklearn package `sklearn.metrics.pairwise import cosine_similarity`, are these two comparable?

```
from sklearn.metrics.pairwise import cosine_similarity
X = [[4.1, 0.5, 1.0]]
Y = [[3.0, 0, 2.5]]
cosine_similarity(X, Y)
```

File: third_party/bigframes_vendored/sklearn/metrics/pairwise.py (hunk 1 above)
Comment (GarrettWu) [reply third_party/bigframes_vendored/sklearn/metrics/pairwise.py:26]: Done.

File: bigframes/ml/core.py (hunk 2 above)
Comment (GarrettWu) [reply bigframes/ml/core.py:65]: Done.

File: tests/system/small/ml/test_metrics_pairwise.py (hunk 3 above)
Comment (GarrettWu) [reply tests/system/small/ml/test_metrics_pairwise.py:22]: Totally different, sklearn takes matrixes and returns matrixes, which we will be hard to support.

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**